alchemiscale_fah/compute/service.py

```diff
@@ -567,47 +563,50 @@ async def execute_DAG(
 
-            # if this is a FahSimulationUnit, then we await its execution in-process
-            if isinstance(unit, FahSimulationUnit):
-                fah_context = FahContext(
-                    shared=shared,
-                    scratch=scratch,
-                    fah_client=fah_client,
-                    fah_projects=fah_projects,
-                    transformation_sk=transformation_sk,
-                    task_sk=task_sk,
-                    index=index,
-                    fah_poll_interval=fah_poll_interval,
-                    encryption_public_key=encryption_public_key,
-                )
-
-                result = await unit.execute(
-                    context=fah_context, raise_error=raise_error, **inputs
-                )
-            else:
-                # otherwise, execute with process pool, allowing CPU bound
-                # units to parallelize across multiple tasks being executed
-                # at once
-
-                # TODO instead of immediately `await`ing here, we could build
-                # up a task for each ProtocolUnit whose deps are satisfied, and
-                # only proceed with additional ones as their deps are satisfied;
-                # would require restructuring this whole method around that
-                # approach, in particular handling retries
-                result = await loop.run_in_executor(
-                    pool,
-                    execute_unit,
-                    json.dumps(
-                        KeyedChain.gufe_to_keyed_chain_rep(unit),
-                        cls=JSON_HANDLER.encoder,
-                    ),
-                    params,
-                )
-
-            all_results.append(result)
-
-            if not keep_scratch:
-                shutil.rmtree(scratch)
+            try:
+                # if this is a FahSimulationUnit, then we await its execution in-process
+                if isinstance(unit, FahSimulationUnit):
+                    fah_context = FahContext(
+                        shared=shared,
+                        scratch=scratch,
+                        fah_client=fah_client,
+                        fah_projects=fah_projects,
+                        transformation_sk=transformation_sk,
+                        task_sk=task_sk,
+                        index=index,
+                        fah_poll_interval=fah_poll_interval,
+                        encryption_public_key=encryption_public_key,
+                    )
+
+                    result = await unit.execute(
+                        context=fah_context, raise_error=raise_error, **inputs
+                    )
+                else:
+                    # otherwise, execute with process pool, allowing CPU bound
+                    # units to parallelize across multiple tasks being executed
+                    # at once
+
+                    # TODO instead of immediately `await`ing here, we could build
```

Comment:
Is this still relevant?

Reply:
Ah good catch! We did exactly what this TODO expressed in this PR. :grin: